tracking/traffic_info.cpp

```diff
@@ -0,0 +1,106 @@
+#include "tracking/traffic_info.hpp"
+
+#include "platform/http_client.hpp"
+
+#include "coding/bit_streams.hpp"
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+#include "coding/writer.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/string.hpp"
+
+#include "private.h"
+
+namespace tracking
+{
+namespace
+{
+char const kTestFileName[] = "traffic_data";
+
+bool ReadRemoteFileAsString(string const & url, string & result)
+{
+  platform::HttpClient request(url);
+  if (request.RunHttpRequest() && request.ErrorCode() == 200)
+  {
+    result = request.ServerResponse();
+    return true;
+  }
+  return false;
+}
+}  // namespace
+
+TrafficInfo::TrafficInfo() {}
+
+bool TrafficInfo::ReceiveTrafficData()
+{
+  if (strlen(TRAFFIC_DATA_BASE_URL) == 0)
+    return false;
+
+  string url = string(TRAFFIC_DATA_BASE_URL) + string(kTestFileName);
+  string contents;
+  if (!ReadRemoteFileAsString(url, contents))
+    return false;
+
+  vector<uint8_t> contentBytes(contents.size());
+  for (size_t i = 0; i < contents.size(); ++i)
+    contentBytes[i] = static_cast<uint8_t>(contents[i]);
+
+  DeserializeTrafficData(contentBytes, m_coloring);
```

Comment:
What if DeserializeTrafficData throws an exception? for example content is corrupted (its length is less than needed) then bitReader.Read throws exception, right?
May be return false if deserialization fails? also, m_coloring can be updated only when deserialization is done to make class with a bit stronger guarantee.